### PR TITLE
new features, three.

### PR DESCRIPTION
**Pull request by 0xadada**
* [x] FEATURE: overlay label can be a component
* [x] FEATURE: overlays can have an `expandSizeBy` property to increase the size of the overlay
* [x] FIX: reversed the order that overlays are added to the DOM to address nested overlay issue.

@ef4 @deverstalmage

### CODE REVIEW

**Comment by 0xadada** [comment]
if either @deverstalmage or @ef4 can debunk why this diff is showing changes that aren't relevant, please add a note, i'll fix and open the PR officially.

**Comment by ef4** [comment]
`labelComponent` is ✅

`expandSizeBy` is  ✅, though I would also point out that sometimes you might not need it because it's also possible to set `padding` on the `.overlay-border` class to customize how the overlay looks. But it's true that this doesn't give exactly the same effect as `expandSizeBy`, if you're trying to render content inside the overlay. (Also, if you experiment with this remember to hard-refresh, because CSS hot reloading won't be detected by ember-overlay's DOM measurement code.)

What is the rationale for `reverseOrder`?

**Comment by 0xadada** [comment]
@ef4 In our usecase, we have deeply nested overlays within overlays, and reversing the order allows users to hover & click on the innermost overlay. Theres probably a better name for it.

**Comment by ef4** [comment]
Ah, so if it fixes that case, it should probably just be the default and doesn't need to become a new option.

**Comment by 0xadada** [comment]
@ef4 @deverstalmage i think thats the right call. I'll do that.

**Comment by 0xadada** [comment]
@ef4 done https://github.com/ef4/ember-overlays/pull/16/commits/cbab8c25402e8cfe5f545263d9a1eb256b65610a

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "ember-overlays",
-  "version": "1.0.5",
+  "version": "1.1.0",
```
Comment (0xadada) [inline package.json]: @ef4 i bumped this, but maybe you want to rev it when you publish the package?

Comment (ef4) [reply package.json]: Yeah, in general it's presumptuous to do this in a feature PR. 😛It's not a big deal, I can adjust when merging.

File: addon/components/overlay-marks.js [hunk 2]
```diff
@@ -7,9 +7,13 @@ export default Component.extend({
   layout,
   tagName: '',
   service: service('ember-overlays'),
-  marks: computed('service.marks', function() {
+  marks: computed('service.marks', 'reverseOrder', function() {
```
Comment (ef4) [inline addon/components/overlay-marks.js]: I think `reverseOrder` is not needed anymore here.

File: addon/styles/addon.css [hunk 3]
```diff
@@ -26,14 +26,14 @@
...
-    top: 2px;
+    top: 0;
     transform-origin: bottom left;
     background-color: #63A9FA;
     color: white;
+    pointer-events: none;
 }
 
 .overlay-border {
-    padding: 10px;
```
Comment (ef4) [inline addon/styles/addon.css]: Does this mean that if people want the existing look they now need to pass `expandSizeBy=10`? If yes, can we make that be the default?

File: addon/styles/addon.css [hunk 4]
```diff
@@ -26,14 +26,14 @@
     padding: 0.2em 1.5em 0em;
     position: absolute;
     white-space: nowrap;
-    top: 2px;
+    top: 0;
```
Comment (ef4) [inline addon/styles/addon.css]: Did this change because something else made it change, or is this a matter of opinion?

File: package.json (hunk 1 above)
Comment (0xadada) [reply package.json]: @ef4 i undid this change.

File: addon/components/overlay-marks.js (hunk 2 above)
Comment (0xadada) [reply addon/components/overlay-marks.js]: @ef4 all set.

File: addon/styles/addon.css (hunk 3 above)
Comment (0xadada) [reply addon/styles/addon.css]: @ef4 good call. all set.

File: addon/styles/addon.css (hunk 4 above)
Comment (0xadada) [reply addon/styles/addon.css]: @ef4 reversed this change.

**Comment by 0xadada** [comment]
@ef4 @deverstalmage Travis tests are passing again ✅, i'm finally happy with this PR, the folks here at @salsify can eliminate our fork once this lands and switch to this repo once the next release goes out.

**Comment by ef4** [comment]
Thanks, this looks ready to go. I added you as a committer here and on NPM, please feel free to merge and release.

**Comment by 0xadada** [comment]
@ef4 great! We're at `1.0.7` now, should i go `1.1.0`?

I'll take care of this on Friday.

**Comment by ef4** [comment]
We could go with 2.0 since it changes the DOM order. That could maybe break
someone if they relied on it.

**Pull request merged**